tfadd/internal/state_to_tpl.go

```diff
@@ -84,9 +86,20 @@ func addAttributes(buf *strings.Builder, stateVal cty.Value, attrs map[string]*t
 			buf.WriteString(strings.Repeat(" ", indent))
 			buf.WriteString(fmt.Sprintf("%s = ", name))
 			tok := hclwrite.TokensForValue(val)
-			if _, err := tok.WriteTo(buf); err != nil {
-				return err
+			// use jsonencode if val is valid json object
+			bs := tok.Bytes()
+			if val.Type().Equals(cty.String) {
+				if jsonStr, err := strconv.Unquote(string(bs)); err == nil && len(jsonStr) > 0 {
+					jsonBytes := []byte(jsonStr)
+					if (jsonBytes[0] == '{' || jsonBytes[0] == '[') && json.Valid(jsonBytes) {
+						var jsonObj interface{}
+						_ = json.Unmarshal(jsonBytes, &jsonObj)
```

Comment:
Can we add the error handling here?

Reply:
It has been checked in the if statement with `json.Valid(jsonBytes)`, so I assume it should not get an error here.

Reply:
Fair, then we can just remove the `_ = ` part? Also it's worth adding a comment to mention why it's safe?